pkg/util/collections/map_utils.go

```diff
@@ -133,3 +133,29 @@ func HasKeyAndVal(root map[string]interface{}, path string, val interface{}) boo
 
 	return valObj == val
 }
+
+// MergeSlices takes two slices and returns a new one representing the union of the originals
+func MergeSlices(first, second []interface{}) []interface{} {
```

Comment:
what I would probably do for this function is to have it take a third parameter, which is a comparison function pointer (`..., equal func(obj1, obj2 interface{})  bool`). This function, which the caller would provide, would handle type-asserting the items to the appropriate type since presumably they'd know what the type is, and returning whether they're considered "equal" or not.  It might also need to return an error if the objs can't be converted to the expected type. Then, in the `MergeSlices` function, you'd just have to remove the intersection between the two lists using the provided equality function